ocaml-lsp-server/docs/ocamllsp/typeEnclosing-spec.md

```diff
@@ -0,0 +1,50 @@
+# Type Enclosing Request
+
+## Description
+
+Merlin has a concept of `type enclosing` that gets the type of ident under the
+cursor. It will highlight the ident and display its type. You can climb the
+typed-tree and display the type of bigger expressions surrounding the cursor. In
+order to keep the request stateless, the manipulation related to growing or
+shrinking enclosings is delegated to the client. This request allows to request
+type enclosing under the cursor and then its surrounding enclosings.
+
+## Client capability
+
+There is no client capability relative to this request.
+
+## Server capability
+
+- property name: `handleTypeEnclosing`
+- property type: `boolean`
+
+## Request
+
+- method: `ocamllsp/typeEnclosing`
+- params:
+
+  ```json
+  {
+    "textDocument": TextDocumentIdentifier,
+    "position": Position,
+    "index": uinteger,
+    "verbosity?": uinteger,
+    "rangeEnd?": Position
```

Comment:
I think we should describe the roles of the parameters here, especially rangeend (we could simply reuse the comments in the design doc if they are up-to-date)

Reply:
Instead of this scheme, I'd recommend the following:

1. Either take a `range` parameter
2. Or a `position` parameter

That way you can stick to standard field names and users can use the convenient `Range.t` type to work with your request.

Reply:
You are right. Fixed in https://github.com/ocaml/ocaml-lsp/pull/1304/commits/89efc55328efe1b68879bbc5b388d913ad63ff0b